Tools/autotest/autotest.py

```diff
@@ -521,39 +447,101 @@ def run_tests(steps):
+    parser.add_option("--list", action='store_true', default=False, help='list the available steps')
+    parser.add_option("--viewerip", default=None, help='IP address to send MAVLink and fg packets to')
+    parser.add_option("--map", action='store_true', default=False, help='show map')
+    parser.add_option("--experimental", default=False, action='store_true', help='enable experimental tests')
+    parser.add_option("--timeout", default=3000, type='int', help='maximum runtime in seconds')
+    parser.add_option("--valgrind", default=False, action='store_true', help='run ArduPilot binaries under valgrind')
+    parser.add_option("--gdb", default=False, action='store_true', help='run ArduPilot binaries under gdb')
+    parser.add_option("--debug", default=False, action='store_true', help='make built binaries debug binaries')
+    parser.add_option("-j", default=None, type='int', help='build CPUs')
+    parser.add_option("--frame", type='string', default=None, help='specify frame type')
 
-lckfile = util.reltopdir('../buildlogs/autotest.lck')
-lck = util.lock_file(lckfile)
-if lck is None:
-    print("autotest is locked - exiting.  lckfile=(%s)" % (lckfile,))
-    sys.exit(0)
+    opts, args = parser.parse_args()
 
-atexit.register(util.pexpect_close_all)
 
-if len(args) > 0:
-    # allow a wildcard list of steps
+    steps = [
+    'prerequisites',
+    'build.All',
+    'build.Binaries',
+    # 'build.DevRelease',
+    'build.Examples',
+    'build.Parameters',
+
+    'build.ArduPlane',
+    'defaults.ArduPlane',
+    'fly.ArduPlane',
+    'fly.QuadPlane',
+
+    'build.APMrover2',
+    'defaults.APMrover2',
+    'drive.APMrover2',
+
+    'build.ArduCopter',
+    'defaults.ArduCopter',
+    'fly.ArduCopter',
+
+    'build.Helicopter',
+    'fly.CopterAVC',
+
+    'build.AntennaTracker',
+
+    'build.ArduSub',
+    'defaults.ArduSub',
+    'dive.ArduSub',
+
+    'convertgpx',
+    ]
+
+    skipsteps = opts.skip.split(',')
+
+    # ensure we catch timeouts
+    signal.signal(signal.SIGALRM, alarm_handler)
+    signal.alarm(opts.timeout)
+
+    if opts.list:
+        for step in steps:
+            print(step)
+        sys.exit(0)
+
+    util.mkdir_p(util.reltopdir('../buildlogs'))
+
+    lckfile = util.reltopdir('../buildlogs/autotest.lck')
+    lck = util.lock_file(lckfile)
+
+    if lck is None:
+        print("autotest is locked - exiting.  lckfile=(%s)" % (lckfile,))
+        sys.exit(0)
+
+    atexit.register(util.pexpect_close_all)
+
     matched = []
-    for a in args:
-        arg_matched = False
-        for s in steps:
-            if fnmatch.fnmatch(s.lower(), a.lower()):
-                matched.append(s)
-                arg_matched = True
-        if not arg_matched:
-            print("No steps matched argument ({})".format(a))
-            sys.exit(1)
+    if len(args) > 0:
+        # allow a wildcard list of steps
+        for a in args:
+            matches = fnmatch.filter(steps, a)
```

Comment:
My bad, I just double checked and you're kinda right, except it's the exact other way around: This will make it case sensitive on linux and insensetive on windows.